cli/context/store/store.go

```diff
@@ -18,15 +18,41 @@ import (
 
 // Store provides a context store for easily remembering endpoints configuration
 type Store interface {
+	StorageInfo
+	Write
+}
+
+// Read provides read-only (without list) access to context data
+type Read interface {
+	GetContextMetadata(name string) (ContextMetadata, error)
+	ListContextTLSFiles(name string) (map[string]EndpointFiles, error)
+	GetContextTLSData(contextName, endpointName, fileName string) ([]byte, error)
+}
+
+// List provides listing of contexts on top of Read interface
+type List interface {
+	Read
```

Comment:
Is Read really useful here? I think you should let the user compose between the different interface, and maybe recreate an object which will be Reader and Lister.